### PR TITLE
Convert StripeMetadataSource to Factory pattern

### PR DESCRIPTION
**Pull request by arunthirupathi**
Inject StripeMetadataSourceFactory instead of
StripeMetadatasource. This PR is in preparation
for introducing DWRF stripe cache.

Depended by https://github.com/facebookexternal/presto-facebook/pull/1684

Test plan 
Existing tests.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by sdruzkin — APPROVED** [review]
LGTM

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java [hunk 1]
```diff
@@ -277,7 +278,7 @@ public OrcFileTailSource createOrcFileTailSource(OrcCacheConfig orcCacheConfig,
 
     @Singleton
     @Provides
-    public StripeMetadataSource createStripeMetadataSource(OrcCacheConfig orcCacheConfig, MBeanExporter exporter)
+    public StripeMetadataSourceFactory createStripeMetadataSourceFactory(OrcCacheConfig orcCacheConfig, MBeanExporter exporter)
```
[the remark this answers is not in the document]
Comment (NikhilCollooru) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java:281]: Ah you just fixed that as well. cool.

[the remark this answers is not in the document]
Comment (arunthirupathi) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java:281]: I forgot my jenkins credentials, dig it out and was able to infer the issue :)

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java [hunk 2]
```diff
@@ -299,7 +300,7 @@ public StripeMetadataSource createStripeMetadataSource(OrcCacheConfig orcCacheCo
             exporter.export(generatedNameOf(CacheStatsMBean.class, connectorId + "_StripeFooter"), footerCacheStatsMBean);
             exporter.export(generatedNameOf(CacheStatsMBean.class, connectorId + "_StripeStream"), streamCacheStatsMBean);
         }
-        return stripeMetadataSource;
+        return StripeMetadataSourceFactory.of(stripeMetadataSource);
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java:303]: Just realized the signature of `of` function is with `src` variable name. Could you submit another PR to quickly fix the nit (`source`)?

**Review by highker — APPROVED** [review]
